app/controllers/application_controller.rb

```diff
@@ -630,6 +630,22 @@ def onboarding_topbar_props
 
   helper_method :onboarding_topbar_props
 
+  def topbar_props
+    {
+      logo: {
+        href: '/',
+        text: @current_community.name(I18n.locale),
+        image: @current_community.wide_logo.present? ? @current_community.wide_logo.url(:header) : nil,
+        image_highres: @current_community.wide_logo.present? ? @current_community.wide_logo.url(:header_highres) : nil
+      },
+      search_mode: 'keyword-and-location',
+      search_keyword_placeholder: 'Search...',
+      search_location_placeholder: 'Location'
```

Comment:
The other point is that with subobject combined props are UI / layout specific, and that should not be decided in backend (whether it's API or Rails)